src/ocean/util/UnixSocketCommandInterface.d

```diff
@@ -0,0 +1,219 @@
+/*******************************************************************************
+
+    Utility for handling requests via a unix socket.
+
+    Commands can be registered to the extension and when the command is received
+    the provided delegate will be called. Any provided arguments will be split
+    by the space character and provided as an array.
+
+    Copyright:
+        Copyright (c) 2009-2018 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.util.UnixSocketCommandInterface;
+
+import ocean.transition;
+import ocean.core.Verify;
+
+import ocean.core.Enforce;
+import ocean.io.select.EpollSelectDispatcher;
+
+/// ditto
+public class UnixSocketCommandInterface
+{
+    import core.sys.posix.sys.stat;
+
+    import ocean.io.select.EpollSelectDispatcher;
+    import ocean.net.server.unix.UnixListener;
+    import ocean.text.util.StringC;
+
+    /// Registry with the supported commands
+    private CommandsRegistry commands_registry;
+
+    /// Unix listener with dynamic command handling.
+    private UnixSocketListener!(CommandsRegistry) unix_listener;
+
+    /// Convenience alias to declare types of command handlers
+    public alias CommandsRegistry.Handler Handler;
+
+    /// Path to create the unix socket.
+    private cstring path;
+
+    /// Permissions mode to apply to the unix socket after binding
+    private int mode = -1;
+
+    /****************************************************************************
+
+        Constructor.
+
+    ****************************************************************************/
+
+    this ( )
+    {
+        this.commands_registry = new CommandsRegistry;
+    }
+
+    /***************************************************************************
+
+        Binds to the path and listens for the new commands
+
+        Params:
+            epoll = Epoll instance.
+            path = path to bind the socket to
+            mode = UNIX permissions mode to apply to the socket after it's
+                   created if needed
+
+    ***************************************************************************/
+
+    public void listen ( EpollSelectDispatcher epoll,
+            cstring path, int mode = -1)
+    {
+        this.path = path;
+        this.mode = mode;
+
+        this.unix_listener =
+            new typeof(this.unix_listener)(this.path, epoll, this.commands_registry);
+
+        if (this.mode >= 0)
```

Comment:
Could it make any sense to move this mode setting into `UnixSocketListener`? If that were done, I think we could do away with `UnixSocketCommandInterface`. Usage would look like:
```D
auto commands = new CommandsRegistry;
auto unix_listener = new UnixListener(this.path, mode, epoll, commands);
epoll.register(unix_listener);
```

Reply:
Yeah good idea, it is a property of unix socket and it should be there.